src/AspNetCoreModuleV2/CommonLib/FileOutputManager.cpp

```diff
@@ -0,0 +1,193 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+#include "stdafx.h"
+#include "FileOutputManager.h"
+#include "sttimer.h"
+#include "exceptions.h"
+#include "debugutil.h"
+#include "SRWExclusiveLock.h"
+#include "file_utility.h"
+#include "StdWrapper.h"
+
+extern HINSTANCE    g_hModule;
+
+FileOutputManager::FileOutputManager() :
+    FileOutputManager(/* fEnableNativeLogging */ true) { }
+
+FileOutputManager::FileOutputManager(bool fEnableNativeLogging) :
+    m_hLogFileHandle(INVALID_HANDLE_VALUE),
+    m_disposed(false),
+    stdoutWrapper(nullptr),
+    stderrWrapper(nullptr),
+    m_fEnableNativeRedirection(fEnableNativeLogging)
+{
+    InitializeSRWLock(&m_srwLock);
+}
+
+FileOutputManager::~FileOutputManager()
+{
+    FileOutputManager::Stop();
+}
+
+HRESULT
+FileOutputManager::Initialize(PCWSTR pwzStdOutLogFileName, PCWSTR pwzApplicationPath)
+{
+    RETURN_IF_FAILED(m_wsApplicationPath.Copy(pwzApplicationPath));
+    RETURN_IF_FAILED(m_wsStdOutLogFileName.Copy(pwzStdOutLogFileName));
+
+    return S_OK;
+}
+
+// Start redirecting stdout and stderr into the file handle.
+// Uses sttimer to continuously flush output into the file.
+HRESULT
+FileOutputManager::Start()
+{
+    SYSTEMTIME systemTime;
+    SECURITY_ATTRIBUTES saAttr = { 0 };
+    STRU struPath;
+
+    // Concatenate the log file name and application path
+    RETURN_IF_FAILED(FILE_UTILITY::ConvertPathToFullPath(
+        m_wsStdOutLogFileName.QueryStr(),
+        m_wsApplicationPath.QueryStr(),
+        &struPath));
+
+    RETURN_IF_FAILED(FILE_UTILITY::EnsureDirectoryPathExist(struPath.QueryStr()));
+
+    // Get the module name and add it to the log file name
+    // as two log files will be created, one from the shim
+    // and one from the request handler.
+    WCHAR path[MAX_PATH];
+    RETURN_LAST_ERROR_IF(!GetModuleFileName(g_hModule, path, sizeof(path)));
+    std::filesystem::path fsPath(path);
+
+    // TODO fix string as it is incorrect
```

Comment:
???

Reply:
Relates to removing module name from log file. Deciding whether to do it in this PR or separate as sharing the same log file can be tricky.